src/init/features/dataconnect/index.ts

```diff
@@ -18,20 +20,159 @@ const SCHEMA_TEMPLATE = readFileSync(join(TEMPLATE_ROOT, "schema.gql"), "utf8");
     type: "input",
-    default: "my-service",
+    name: "localConnectionString",
+    message: `What is the connection string of the local Postgres instance you would like to use with the Data Connect emulator?`,
+    default: defaultConnectionString,
   });
+  setup.rcfile.dataconnectEmulatorConfig = { postgres: { localConnectionString } };
+
+  const dir: string = config.get("dataconnect.source") || "dataconnect";
+  const subbedDataconnectYaml = subValues(DATACONNECT_YAML_TEMPLATE, info);
+  const subbedConnectorYaml = subValues(CONNECTOR_YAML_TEMPLATE, info);
+
+  await config.askWriteProjectFile(join(dir, "dataconnect.yaml"), subbedDataconnectYaml);
+  await config.askWriteProjectFile(join(dir, "schema", "schema.gql"), SCHEMA_TEMPLATE);
+  await config.askWriteProjectFile(
+    join(dir, info.connectorId, "connector.yaml"),
+    subbedConnectorYaml,
+  );
+  await config.askWriteProjectFile(join(dir, info.connectorId, "queries.gql"), QUERIES_TEMPLATE);
+  await config.askWriteProjectFile(
+    join(dir, info.connectorId, "mutations.gql"),
+    MUTATIONS_TEMPLATE,
+  );
+  if (
+    setup.projectId &&
+    (info.isNewInstance || info.isNewDatabase) &&
+    (await confirm({
+      message:
+        "Would you like to provision your CloudSQL instance and database now? This will take a few minutes.",
+      default: true,
+    }))
+  ) {
+    await provisionCloudSql({
+      projectId: setup.projectId,
+      locationId: info.locationId,
+      instanceId: info.cloudSqlInstanceId,
+      databaseId: info.cloudSqlDatabase,
+      enableGoogleMlIntegration: false,
+    });
+  }
+}
+
+function subValues(
+  template: string,
+  replacementValues: {
+    serviceId: string;
+    cloudSqlInstanceId: string;
+    cloudSqlDatabase: string;
+    connectorId: string;
+  },
+): string {
+  const replacements: Record<string, string> = {
+    serviceId: "__serviceId__",
+    cloudSqlDatabase: "__cloudSqlDatabase__",
+    cloudSqlInstanceId: "__cloudSqlInstanceId__",
+    connectorId: "__connectorId__",
+  };
+  let replaced = template;
+  for (const [k, v] of Object.entries(replacementValues)) {
+    replaced = replaced.replace(replacements[k], v);
+  }
+  return replaced;
+}
 
-  let cloudSqlInstanceId = "";
-  let newInstance = false;
-  let locationId = "";
+async function promptForService(setup: Setup, info: RequiredInfo): Promise<RequiredInfo> {
   if (setup.projectId) {
+    await ensureApis(setup.projectId);
+    // TODO: Support initing with services that have existing sources/files
+    const existingServices = await listAllServices(setup.projectId);
+    const existingServicesAndSchemas = await Promise.all(
+      existingServices.map(async (s) => {
+        return {
+          service: s,
+          schema: await getSchema(s.name),
+        };
+      }),
+    );
+    const existingFreshServicesAndSchemas = existingServicesAndSchemas.filter((s) => {
+      return !s.schema?.source.files?.length;
+    });
```

Comment:
Oh, very interesting, so don't even include not fresh services in the prompt.

Shall we add a TODO to download not fresh services as well?

Reply:
Yep - that was the intention of the TODO on L123